lib/alerts/alert.ex

```diff
@@ -137,9 +137,37 @@ defmodule Alerts.Alert do
     alert
     |> struct!(keywords)
     |> set_priority()
+    |> set_direction_ids()
     |> ensure_entity_set()
   end
 
+  defp set_direction_ids(%__MODULE__{effect: :station_closure} = alert) do
+    informed_entities = Enum.map(alert.informed_entity, &set_direction_id/1)
+
+    Map.put(alert, :informed_entity, informed_entities)
+  end
+
+  defp set_direction_ids(%__MODULE__{} = alert), do: alert
+
+  defp set_direction_id(%InformedEntity{direction_id: nil, route_type: 1} = entity) do
+    stop = Stops.Repo.get(entity.stop)
+
+    if stop.child_ids == [] do
+      direction_id =
+        stop.id
+        |> RoutePatterns.Repo.by_stop_id()
```

Comment:
let's get the repo module via `Application.compile_env!(:dotcom, :repo_modules)[:route_patterns]`